### DISCUSSION TITLE
Use for "kiosk" mode devices: prevent access to admin settings

### DISCUSSION THREAD

**Discussion by mtx-z** [AnySoftKeyboard/AnySoftKeyboard#2878] [Ideas]
Hello,

first, this keyboard is truly awesome. It may be the most customizable one and is exactly what I was looking for: an awesome job. 
I would like to use this keyboard on "kiosk" mode devices (tablet): the user can leave a rating in a restaurant. 

So it needs to be "stupid"/"pseudo hacker" proof. Meaning I would like to prevent all settings/admin access from it:
- On the emoji panel, we can access settings on the right of the panel: clicking the screw icon
- On the long "enter" button click, we can also open the settings. 

I would really love an option to disable all admin access from "frontend", so it can only be accessed through android admin settings. 

Thanks for this great job!

**Comment by menny** [comment]
I like this...

this is possible in two ways:
1) you can remove the code related to that and recompile a version for your tablets - you can do this right now.
2) we will add an option in the settings that disable settings access from the keyboard layout - you will need to wait till we have enough bandwidth to do so.

What do you think?

**Comment by mtx-z** [comment]
Hello,

in the end, we forked the repo and customized it to our needs. 

Our objectives are: 
- make the keyboard installation as fast as possible (so teams that will prepare/provisionning hundreds of tablets can do it faster)
   - for example, we integrated the french APK into the main APK so only 1 APK is needed to install
   - we stored the XML config file on our tablet when provisioning, and changed the import path for the file in the keyboard source code. Then, we made the file "auto import" to reduce even more the click number needed to install and setup the keyboard for our provisioning teams. 
- prevent user to leave the keyboard to android, or voice input settings, clipboard settings, etc... nothing but the keyboard, and no click/long press/swipe or anything can make the user leave the actual app, or open a popup. User is "locked" in our app with our MDM app, and so the keyboard must not be a way outside. 
- auto enable "private mode" so no history is kept
- make it "idiot-proof": close emoji keyboard after first input (so next user doesn't come to an emoji keyboard)

Here's the repo: https://github.com/mtx-z/AnySoftKeyboard-Liz 

I added in the readme a task list we did. 

In the end, I think that with some additional options, AnySoftKeyboard can be the best (and only) android keyboard usable for Kiosk mode android devices. I tested 20 or 30 keyboards, AnySoft one is the only one we can customize enough with provided options, and open source so we can tweak it to perfection :) 

I can provide you with the "steps" given to our team to provision tablets (keyboard steps) so you can see how fast we made it. If you want, we can also talk about the tweaks we made, and what is the one worth to be made as a keyboard setting. 

Thanks a lot anyway for the great work!

**Reply by menny** [reply]
this is great! I'm very glad you solved this.

Please consider creating PRs for things you think will help AnySoftKeyboard
